frontend/taipy-gui/src/components/Taipy/Metric.spec.tsx

```diff
@@ -11,21 +11,415 @@
+                [0.5555555555555556, "#d8576b"],
+                [0.6666666666666666, "#ed7953"],
+                [0.7777777777777778, "#fb9f3a"],
+                [0.8888888888888888, "#fdca26"],
+                [1, "#f0f921"]
+            ]
+        },
+        paper_bgcolor: "rgb(31,47,68)",
+    }
+}
+
+const lightTemplate = {
+    layout: {
+        colorscale: {
+            diverging: [
+                [0, "#053061"],
+                [0.1, "#2166ac"],
+                [0.2, "#4393c3"],
+                [0.3, "#92c5de"],
+                [0.4, "#d1e5f0"],
+                [0.5, "#f7f7f7"],
+                [0.6, "#fddbc7"],
+                [0.7, "#f4a582"],
+                [0.8, "#d6604d"],
+                [0.9, "#b2182b"],
+                [1, "#67001f"]
+            ],
+            sequential: [
+                [0, "#f7fcf5"],
+                [0.1111111111111111, "#e5f5e0"],
+                [0.2222222222222222, "#c7e9c0"],
+                [0.3333333333333333, "#a1d99b"],
+                [0.4444444444444444, "#74c476"],
+                [0.5555555555555556, "#41ab5d"],
+                [0.6666666666666666, "#238b45"],
+                [0.7777777777777778, "#006d2c"],
+                [0.8888888888888888, "#00441b"],
+                [1, "#000000"]
+            ],
+            sequentialminus: [
+                [0, "#f7fcf5"],
+                [0.1111111111111111, "#e5f5e0"],
+                [0.2222222222222222, "#c7e9c0"],
+                [0.3333333333333333, "#a1d99b"],
+                [0.4444444444444444, "#74c476"],
+                [0.5555555555555556, "#41ab5d"],
+                [0.6666666666666666, "#238b45"],
+                [0.7777777777777778, "#006d2c"],
+                [0.8888888888888888, "#00441b"],
+                [1, "#000000"]
+            ]
+        },
+        paper_bgcolor: "rgb(255,255,255)",
+    }
+}
+
 describe("Metric Component", () => {
-  it("renders", async () => {
-    const { getByTestId } = render(<Metric testId="test-id"/>);
-    const elt = getByTestId("test-id");
-    expect(elt.tagName).toBe("DIV");
-  })
-  it("displays the right info for class", async () => {
-    const { getByTestId } = render(<Metric testId="test-id" className={'taipy-gauge'}/>);
-    const elt = getByTestId('test-id');
-    expect(elt).toHaveClass('taipy-gauge');
-  })
+    it("renders", async () => {
+        const {getByTestId} = render(<Metric testId="test-id"/>);
+        const elt = getByTestId("test-id");
+        expect(elt.tagName).toBe("DIV");
+    })
+
+    it("displays the right info for class", async () => {
+        const {getByTestId} = render(<Metric testId="test-id" className={'taipy-gauge'}/>);
+        const elt = getByTestId('test-id');
+        expect(elt).toHaveClass('taipy-gauge');
+    })
+
+    it("sets the title when provided", async () => {
+        const title = "Test Title";
+        let container: RenderResult | null = null;
+        container = render(<Metric title={title}/>);
```

Comment:
why not use const {container}=render(...) ?

Reply:
Updated